### PR TITLE
refactor(abstract): Use abstract keyword where possible to decrease f…

### PR DESCRIPTION
**Pull request by mhevery**
…ile size.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: modules/@angular/compiler/src/config.ts [hunk 1]
```diff
@@ -52,12 +48,12 @@ export class CompilerConfig {
  * to help tree shaking.
  */
 export abstract class RenderTypes {
-  get renderer(): CompileIdentifierMetadata { return unimplemented(); }
-  get renderText(): CompileIdentifierMetadata { return unimplemented(); }
-  get renderElement(): CompileIdentifierMetadata { return unimplemented(); }
-  get renderComment(): CompileIdentifierMetadata { return unimplemented(); }
-  get renderNode(): CompileIdentifierMetadata { return unimplemented(); }
-  get renderEvent(): CompileIdentifierMetadata { return unimplemented(); }
+  abstract get renderer(): CompileIdentifierMetadata;
```
Comment (vicb) [inline modules/@angular/compiler/src/config.ts]: great, this is new in TS, right - IIRC abstract getter where not supported before ?

**Comment by vicb** [comment]
LGTM if LGTTravis

File: modules/@angular/compiler/src/config.ts (hunk 1 above)
Comment (mhevery) [inline modules/@angular/compiler/src/config.ts]: yes this is new.

**Comment by jasonaden** [comment]
LGTM

**Comment by vicb** [comment]
LGTM

**Review by tbosch — APPROVED** [review]

**Comment by mhevery** [comment]
Merged by: https://github.com/angular/angular/commit/670b680b0a510267142c38fe1f487a180da74eba

**Comment by alan-agius4** [comment]
How does using 'abstract' reduce the bundle size please?

**Comment by tytskyi** [comment]
@alan-agius4 they are not present in generated code

**Pull request closed without merging**